### PR TITLE
Add typechain for smart contract types and minor fixes

### PR DESCRIPTION
**Pull request by Granahh**
I have added typechain, it is free because you have it as a dependency already added and I find it much more convenient as a developer experience. I have also fixed some little things.

MRC to the moon 🚀!

### CODE REVIEW

**Review by InfoLorenzo — APPROVED** [review]
LGTM

**Comment by kingb1rd** [comment]
LGTM

File: test/Basic.ts [hunk 1]
```diff
@@ -2,19 +2,19 @@ import dotenv from 'dotenv'
 import { expect } from "chai";
 import hre from "hardhat";
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
-import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
-
+import { E7LBasic } from "../typechain-types";
+import { IMRC } from "../typechain-types";
```
Comment (DanielSintimbrean) [inline test/Basic.ts:6]:
```suggestion
import { IMRC, E7LBasic } from "../typechain-types";
```

File: test/Reverts.ts [hunk 2]
```diff
@@ -2,6 +2,8 @@ import dotenv from 'dotenv'
 import { expect } from "chai";
 import hre from "hardhat";
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { E7LBasic } from "../typechain-types";
+import { IMRC } from "../typechain-types";
```
Comment (DanielSintimbrean) [inline test/Reverts.ts:6]:
```suggestion
import { E7LBasic, IMRC } from "../typechain-types";
```

File: test/Basic.ts (hunk 1 above)
Comment (Granahh) [reply test/Basic.ts]: Done! Ty!

File: test/Reverts.ts (hunk 2 above)
Comment (Granahh) [reply test/Reverts.ts]: Done! Ty!

**Comment by Granahh** [comment]
Guys, maybe you need to update this file https://github.com/Racks-Labs/ERC721-Linkable/blob/main/.github/workflows/ci.yml and add a new step that can run this command -> `npm run typechain`

**Comment by DanielSintimbrean** [comment]
Seem to be the secrets for environment variables 

image

This is the last one that run without errors
image

**Comment by Granahh** [comment]
Yep @DanielSintimbrean `TypeError [ERR_INVALID_URL]: Invalid URL`

Have to check the secrets environment in the project configuration, but I add a new commit to create this step, I think it is important to generate the types before the unit test step.

**Comment by DanielSintimbrean** [comment]
image

**Comment by Granahh** [comment]
> image

https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Pull request merged**